src/NodeParser/InterfaceAndClassNodeParser.ts

```diff
@@ -96,7 +97,27 @@ export class InterfaceAndClassNodeParser implements SubNodeParser {
         return node.heritageClauses.reduce(
             (result: BaseType[], baseType) => [
                 ...result,
-                ...baseType.types.map((expression) => this.childNodeParser.createType(expression, context)),
+                ...baseType.types
+                    .map((expression) => {
+                        // Skip processing of TypeScript lib utility types in heritage clauses
+                        // to avoid infinite recursion with recursive types
+                        const typeSymbol = this.typeChecker.getSymbolAtLocation(expression.expression);
+                        if (typeSymbol && typeSymbol.flags & ts.SymbolFlags.Alias) {
+                            const aliasedSymbol = this.typeChecker.getAliasedSymbol(typeSymbol);
+                            // Check if any declaration is from a TypeScript lib file
+                            // Lib utility types (Omit, Pick, etc.) should be skipped to prevent
```

Comment:
Let's test that omit and pick if not recursive work well even after this change.